hips/hip-0019.md

```diff
@@ -0,0 +1,46 @@
+---
+hip: 0019
+title: "Add Exclude File Option to Helm Lint Command"
+authors: Danilo Patrucco, Daniel Pritchett
+created: "2024-07-03"
+type: "feature"
+status: "draft"
+---
+
+## Abstract 
+This proposal suggests enhancing the `helm lint` command to exclude specific files or patterns from linting through a `.helmlintignore` file, similar to `.gitignore` or `.helmignore`. Additionally, a `--lint-ignore-file` flag will be introduced to specify alternative locations for the `.helmlintignore` file, facilitating its use in projects with multiple sub-charts. 
+## Motivation 
+In large Helm charts, certain files or configurations may deviate from standard linting rules but are accepted by the maintainers. Currently, `helm lint` evaluates all files within a chart, leading to irrelevant warnings or errors. An exclusion option and the ability to specify ignore file locations would allow developers to better manage lint results. 
+## Rationale 
+This proposal aims to give developers and integrators greater control over the linting process by allowing exclusions for: 
+- Files imported or managed by third parties that are not directly editable. 
+- Files that may not conform to static linting rules. 
+- Files containing intentional deviations from standard practices due to specific deployment scenarios. 
+The `.helmlintignore` file will support simple patterns to match files and directories for easy exclusion management. The `--lint-ignore-file` flag enhances this by allowing centralized management of lint exclusions in complex projects. 
+## Specification 
```

Comment:
Also: The lint rules/config should be packaged with the Chart. As `helm lint` (and `helm lint --with-subcharts`) will want to be able to include the config when linting charts pull from chart repositories/registries.

ie. we want this to work:
```
helm pull <url> 
helm lint <chart>.tgz
```

Without packaging the lint config with the chart, it would only be possible to have the config apply when checking the chart out from source (ie. git repo)

And this will certainly prevent lint config from being utilized when linting subcharts (the `--with-subcharts` above)

Reply:
currently the --with-subchart command seems to be broken when a chart is packaged, see discussion in helm-dev slack:

[helm-dev slack discussion](https://kubernetes.slack.com/archives/C51E88VDG/p1726696306443959)

Reply:
also, we verified the the `.helmlintconfig.yaml` file would be packaged with the chart